code/src/healthstatus/checker.py

```diff
@@ -198,11 +198,16 @@ async def test_random_outdated_asset_first(self) -> bool:
         # Returns True if anything tested
         all_assets = [asset async for asset in self.aiterassets()]
         all_asset_paths = {asset.asset_path for asset in all_assets}
-        if outdated := (self.reporter.outdated_assets() & all_asset_paths) | (
+        outdated_paths = (self.reporter.outdated_assets() & all_asset_paths) | (
             all_asset_paths - self.reporter.all_assets()
-        ):
-            p = choice(list(outdated))
-            asset = Asset(filepath=self.mount_path / p, asset_path=p)
+        )
+        outdated = [
+            asset
+            for p in outdated_paths
+            if (asset := Asset(filepath=self.mount_path / p, asset_path=p)).is_nwb()
```

Comment:
note: at some point we should add tests for other files as well, but ATM it is ok